test/Options.ts

```
@@ -348,4 +349,44 @@ describe.concurrent("Options", () => {
       const result = yield* $(Options.validate(option, args, config))
       expect(result).toEqual([["--verbose"], HashMap.make(["key1", "v1"], ["key2", "v2"])])
     }))
+
+  it.effect("variadic - invalid integer", () =>
+    Effect.gen(function*($) {
+      const config = CliConfig.defaultConfig
+      const option = Options.alias(Options.repeat1(Options.integer("defs")), "d")
+      const args = ["-d", "1", "-d", "v2", "-d", "3", "--verbose"]
+      const error = yield* $(Effect.flip(Options.validate(option, args, config)))
+      expect(error).toEqual(ValidationError.invalidValue(HelpDoc.p(
+        `"v2" is not a integer`
+      )))
+    }))
+
+  it.effect("variadic - missing value", () =>
+    Effect.gen(function*($) {
+      const config = CliConfig.defaultConfig
+      const option = Options.alias(Options.repeat1(Options.integer("defs")), "d")
+      const args = ["--verbose"]
+      const error = yield* $(Effect.flip(Options.validate(option, args, config)))
+      expect(error).toEqual(ValidationError.missingValue(HelpDoc.p(Span.error(
+        "Expected to find option: '--defs'"
+      ))))
+    }))
+
+  it.effect("variadic - integers", () =>
+    Effect.gen(function*($) {
+      const config = CliConfig.defaultConfig
+      const option = Options.alias(Options.repeat1(Options.integer("defs")), "d")
+      const args = ["-d", "1", "-d", "2", "--verbose"]
+      const result = yield* $(Options.validate(option, args, config))
+      expect(result).toEqual([["--verbose"], Chunk.make(1, 2)])
+    }))
+
+  it.effect("variadic - integers atMost 2", () =>
```

@IMax153 Not sure if we actually want to ignore extra options, or raise an error.

---

I think if we're specifying something like `atMost`, we should probably raise an error.